lib/src/sip_ua_helper.dart

```diff
@@ -146,55 +162,61 @@ class SIPUAHelper extends EventManager {
 
     eventHandlers.on(EventConnecting(), (EventConnecting event) {
       logger.debug('call connecting');
-      _handleCallState('connecting', null, null, null, null, null);
+      _notifyCallStateListeners(CallState(CallStateEnum.CONNECTING));
     });
 
     eventHandlers.on(EventProgress(), (EventProgress event) {
       logger.debug('call is in progress');
-      _handleCallState(
-          'progress', event.response, null, event.originator, null, null);
+      _notifyCallStateListeners(
+          CallState(CallStateEnum.PROGRESS, originator: event.originator));
     });
 
     eventHandlers.on(EventFailed(), (EventFailed event) {
       logger.debug('call failed with cause: ' + (event.cause));
-      _handleCallState(
-          'failed', event.response, null, event.originator, null, null);
+      _notifyCallStateListeners(
+          CallState(CallStateEnum.FAILED, originator: event.originator));
```

Comment:
Same as registration, Need a way to send `code` and `cause` to the UI, telling why the call failed.